### ISSUE TITLE
Properly update cosmetic colour on team switch

### ISSUE THREAD

**Issue by azzyr** [tf2classic/tf2classic-issue-tracker#39]
Changing team while having a team-coloured item equipped prevents the colour of the item updating to the new team's colour

Colour only updates if you unequip the item, resupply, and equip it again

**Comment by Yakibomb** [comment]
Yeah this bug is super prominent in the medieval pack Renaissance Revival I just released. I really hope this gets fixed, otherwise the pack will be completely unplayable in the current version of TF2C.
